### PR TITLE
Add increment proto build and use in CI

### PR DESCRIPTION
**Pull request by dabacon**
Adds a script to only build changed proto files in `cirq/google/api`.

Updates Google Actions to use this script instead of the full build.

Currently includes two changed proto files to test.  Will revert these before submission. 

Could have also made one script but opted for separate. Feedback welcome :) 

Also updates the full build tool to automatically grab all protos in `cirq/google/api` instead of having a hard coded list.

Fixes some documentation.

### CODE REVIEW

**Comment by dabacon** [comment]
Fixes #2360

**Comment by dabacon** [comment]
Fixes #2360

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -129,8 +129,8 @@ jobs:
           python -m pip install -r dev_tools/conf/pip-list-dev-tools.txt
           wget https://github.com/bazelbuild/bazel/releases/download/0.26.0/bazel_0.26.0-linux-x86_64.deb
           sudo dpkg -i bazel_0.26.0-linux-x86_64.deb
-      - name: Build protos
-        run: check/build-protos
+      - name: Build changed protos
+        run: check/build-changed-protos
```
Comment (Strilanc) [inline .github/workflows/ci.yml:133]: We're positive we can't break the proto build except by editing the protos themselves? For example, if we update the build-changed-protos file but not the protos themselves...

Comment (dabacon) [reply .github/workflows/ci.yml:133]: This is a goodpoint.  If the build files are changed, then I think we should build the corresponding proto.

Comment (dabacon) [reply .github/workflows/ci.yml:133]: PTAL.  Added that it builds all protos if the BUILD rule has changed.

File: check/build-changed-protos [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+if [[ ! -z "$changed" ]]; then
+    for base in $changed:
+    do
+        echo -e "Found changed proto: '${base}.proto'"
+    done
+fi
+
+# Check if any of the build files have changed.
+build_changed=$(git diff --name-only ${rev} -- \
+    | grep "google/api/.*BUILD$" \
```
Comment (Strilanc) [inline check/build-changed-protos:76]: Also this script file itself could change.

**Comment by Strilanc** [comment]
The logic in this file is getting quite complicated.  Check out `bash_scripts_test.py` if you want to test it.

**Comment by dabacon** [comment]
The replaced Github action passes, but for some reason the status check is not reporting it as so.

**Comment by dabacon** [comment]
This is stuck on the old action not being reported.  Is there further config that needs to be done for actions?  May try renaming it to old name.

**Pull request merged**